assets/src/dashboard/components/storyAnimation/AMPKeyframes.js

```diff
@@ -14,29 +14,50 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import { useMemo } from 'react';
+
 /**
  * Internal dependencies
  */
-import { KeyframesOutput } from '../../animations/animationOutputs';
-import { AnimationPart } from '../../animations/animationParts';
+import { KeyframesOutput } from '../../animations/outputs';
 import useStoryAnimationContext from './useStoryAnimationContext';
 
+export function generateKeyframesMap(targets, getAnimationParts) {
+  return targets.reduce((acc, target) => {
+    return {
+      ...acc,
+      ...getAnimationParts(target).reduce((a, part) => {
+        const { generatedKeyframes } = part;
+        return {
+          ...a,
+          ...generatedKeyframes,
+        };
+      }, acc),
+    };
+  }, {});
+}
```

Comment:
I don't think we're supporting IE but just to be safe I'll keep things as is only because I don't see `flatMap` used anywhere else in our repo, but it's good to know of a cleaner way of doing this.  My brain tingled that there existed a new js feature to do the flattening but I didn't look it up and stuck with this approach, but thanks for answering my internal thought bubble question for me :D